File: java/src/main/java/com/example/PackVo.java
package com.example;

import java.util.*;

public class PackVo {
    public enum Result {
        SUCCESS, FAILED_1, FAILED_2, FAILED_3;
        static final Result[] ARRAY = new Result[]{ SUCCESS, FAILED_1, FAILED_2, FAILED_3 };
    }

    public static class Category  {
        String name;
        int level;
        long i_column;
        double d_column;
        String des;
        Category[] sub_category;

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (!(o instanceof Category)) return false;
            Category category = (Category) o;
            return level == category.level &&
                    i_column == category.i_column &&
                    Double.compare(category.d_column, d_column) == 0 &&
                    Objects.equals(name, category.name) &&
                    Objects.equals(des, category.des) &&
                    Arrays.equals(sub_category, category.sub_category);
        }
    }

    public static class Data {
        boolean d_bool;
        float d_float;
        double d_double;
        String string_1;
        int int_1;
        int int_2;
        int int_3;
        int int_4;
        int int_5;
        long long_1;
        long long_2;
        long long_3;
        long long_4;
        long long_5;
        Category d_category;
        boolean[] bool_array;
        int[] int_array;
        long[] long_array;
        float[] float_array;
        double[] double_array;
        String[] string_array;

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (!(o instanceof Data)) return false;
            Data data = (Data) o;
            return d_bool == data.d_bool &&
                    Float.compare(data.d_float, d_float) == 0 &&
                    Double.compare(data.d_double, d_double) == 0 &&
                    int_1 == data.int_1 &&
                    int_2 == data.int_2 &&
                    int_3 == data.int_3 &&
                    int_4 == data.int_4 &&
                    int_5 == data.int_5 &&
                    long_1 == data.long_1 &&
                    long_2 == data.long_2 &&
                    long_3 == data.long_3 &&
                    long_4 == data.long_4 &&
                    long_5 == data.long_5 &&
                    Objects.equals(string_1, data.string_1) &&
                    Objects.equals(d_category, data.d_category) &&
                    Arrays.equals(bool_array, data.bool_array) &&
                    Arrays.equals(int_array, data.int_array) &&
                    Arrays.equals(long_array, data.long_array) &&
                    Arrays.equals(float_array, data.float_array) &&
                    Arrays.equals(double_array, data.double_array) &&
                    Arrays.equals(string_array, data.string_array);
        }
    }

    public static class Response {
        Result code;
        String detail;
        Data[] data;

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (!(o instanceof Response)) return false;
            Response response = (Response) o;
            return code == response.code &&
                    Objects.equals(detail, response.detail) &&
                    Arrays.equals(data, response.data);
        }
    }
}

File: Packable-Kotlin/src/main/kotlin/example/PackVo.kt
package example

@Suppress("EqualsOrHashCode", "PropertyName")
class PackVo {
    enum class Result {
        SUCCESS,
        FAILED_1,
        FAILED_2,
        FAILED_3;

        companion object {
            val ARRAY = arrayOf(SUCCESS, FAILED_1, FAILED_2, FAILED_3)
        }
    }

    class Category(
        var name: String?,
        var level: Int,
        var i_column: Long,
        var d_column: Double,
        var des: String? = null,
        var sub_category: Array<Category?>?
    ) {
        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false
            other as Category
            if (name != other.name) return false
            if (level != other.level) return false
            if (i_column != other.i_column) return false
            if (d_column != other.d_column) return false
            if (des != other.des) return false
            if (sub_category != null) {
                if (other.sub_category == null) return false
                if (!sub_category.contentEquals(other.sub_category)) return false
            } else if (other.sub_category != null) return false
            return true
        }
    }

    class Data {
        var d_bool = false
        var d_float = 0f
        var d_double = 0.0
        var string_1: String? = null
        var int_1 = 0
        var int_2 = 0
        var int_3 = 0
        var int_4 = 0
        var int_5 = 0
        var long_1: Long = 0
        var long_2: Long = 0
        var long_3: Long = 0
        var long_4: Long = 0
        var long_5: Long = 0
        var d_category: Category? = null
        var bool_array: BooleanArray? = null
        var int_array: IntArray? = null
        var long_array: LongArray? = null
        var float_array: FloatArray? = null
        var double_array: DoubleArray? = null
        var string_array: Array<String>? = null

        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false

            other as Data

            if (d_bool != other.d_bool) return false
            if (d_float != other.d_float) return false
            if (d_double != other.d_double) return false
            if (string_1 != other.string_1) return false
            if (int_1 != other.int_1) return false
            if (int_2 != other.int_2) return false
            if (int_3 != other.int_3) return false
            if (int_4 != other.int_4) return false
            if (int_5 != other.int_5) return false
            if (long_1 != other.long_1) return false
            if (long_2 != other.long_2) return false
            if (long_3 != other.long_3) return false
            if (long_4 != other.long_4) return false
            if (long_5 != other.long_5) return false
            if (d_category != other.d_category) return false
            if (bool_array != null) {
                if (other.bool_array == null) return false
                if (!bool_array.contentEquals(other.bool_array)) return false
            } else if (other.bool_array != null) return false
            if (int_array != null) {
                if (other.int_array == null) return false
                if (!int_array.contentEquals(other.int_array)) return false
            } else if (other.int_array != null) return false
            if (long_array != null) {
                if (other.long_array == null) return false
                if (!long_array.contentEquals(other.long_array)) return false
            } else if (other.long_array != null) return false
            if (float_array != null) {
                if (other.float_array == null) return false
                if (!float_array.contentEquals(other.float_array)) return false
            } else if (other.float_array != null) return false
            if (double_array != null) {
                if (other.double_array == null) return false
                if (!double_array.contentEquals(other.double_array)) return false
            } else if (other.double_array != null) return false
            if (string_array != null) {
                if (other.string_array == null) return false
                if (!string_array.contentEquals(other.string_array)) return false
            } else if (other.string_array != null) return false

            return true
        }
    }

    class Response(
        val code: Result,
        val detail: String,
        val data: Array<Data>?
    ) {
        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false

            other as Response

            if (code != other.code) return false
            if (detail != other.detail) return false
            if (data != null) {
                if (other.data == null) return false
                if (!data.contentEquals(other.data)) return false
            } else if (other.data != null) return false

            return true
        }
    }
}
